app/api/tool_call/route.tsx

```diff
@@ -0,0 +1,53 @@
+import { AI_MODEL } from "@/lib/consts";
+import { NextRequest } from "next/server";
+import OpenAI from "openai";
+
+export async function POST(req: NextRequest) {
+  const body = await req.json();
+  const context = body.context;
+  const question = body.question;
+
+  try {
+    const openai = new OpenAI();
+
+    const response = await openai.chat.completions.create({
+      model: AI_MODEL,
+      messages: [
+        {
+          role: "user",
+          content: `Context: ${JSON.stringify(context)}
+          Question: ${question}
+          Answer: ????
+
+          Based on provided data, let me know the only answer.
+          If question is related to listening habits, you should provide a broad overview of listening habits based on provided fans' information.
+          In this case, make answer richer and more flexible by providing information about the fans' names, countries, cities, and playlists, albums, episodes, audio books, shows, tracks, and artists that are related to fans.
+
+          You should focus on providing recommendations along with metrics while making answer.
+          For example:
+          "To boost this further, consider running a limited-time exclusive content drop or fan contest"
+          `,
+        },
+        {
+          role: "system",
+          content: `Let's just get a response in only string format without any greeting or useless text. Not JSON format!!!!!!!!!. Don't include any special characters.`,
```

Comment:
Could you please throw these instructions into `cursor` or `chatgpt` for a quick cleanup?
- prompt: `could you please cleanup these instructions to be easier to read?`